app/hotels/src/navigation/AllHotelsNavigationScreen.js

```diff
@@ -59,8 +59,26 @@ type NavigationProps = {|
 
 type Props = ContainerProps & StateProps & DispatchProps & NavigationProps;
 
-export class AllHotelsNavigationScreen extends React.Component<Props> {
+let backButtonListner = null;
+
+export const registerBackButtonListener = (onBackClicked: Function) => {
+  if (Platform.OS === 'android' && backButtonListner === null) {
+    backButtonListner = BackHandler.addEventListener(
+      'hardwareBackPress',
+      () => {
+        onBackClicked();
+        return true;
+      },
+    );
+  }
+};
+
+// Needed for test purposes
```

Comment:
It's almost always wrong to do something like this. Tests should guide you towards better application design but it doesn't mean you should tweak application code in order do be able to test it in this ugly way.. 🙂